library/llm.py

```diff
@@ -6,28 +6,65 @@ def __init__(self, model_name, vdb: VDB):
         self.model = GPT4All(model_name)
         self.vdb = vdb
 
-    def query(self, question, max_tokens=50):
+    def query(self, question, key, context_limit = 5, max_tokens=500):
 
-        print("Asking   : ", question)
-        context = self.vdb.search(question)
+        context = self.vdb.search(question, key, context_limit)
         
-        print("Context  : ", str(context))
-        
-        # .join(" | ")
+        emails = []
+        for o in context.objects:
+            emails.append(o.properties['text'])
+            
+        mail_context = '"' + '"\n\n"'.join(emails) + '"\n\n'
+
+        print("Retrieving" + str(len(emails)) + ' emails')
+        print("Context " + mail_context)
 
         prompt = """
+        ### Instruction:
```

Comment:
There is a format expected by these models as a matter of configuration. This conforms to it so we get better results